### DISCUSSION TITLE
How does !important and Media Query work together?

### DISCUSSION THREAD

**Discussion by guangzan** [windicss/windicss#387] [Q&A]
![image](https://user-images.githubusercontent.com/43458667/125029582-7832ca00-e0bc-11eb-8df0-ea860d82d909.png)
It doesn't work .

**Comment by hannoeru** [comment]
Put !important mark before media query `!<md:w-full`

**Reply by guangzan** [reply]
Thanks for your reply. I've tried `!<md:w-full`  `<md:!w-full`  `<md:(!w-full)`， but they don't work.

**Reply by hannoeru** [reply]
~I think is vite-plugin-windicss, string include "<" will not works now, maybe you can use something like `w-full md:w-auto`~

looks like they already fixed it https://github.com/windicss/vite-plugin-windicss/issues/195 can you upgrade to the latest version?

**Reply by guangzan** [reply]
<s>I solved it, just add a space between `<md:`  and `!w-full` to get `<md: !w-full`.</s>

**Reply by guangzan** [reply]
It's not right ...

**Reply by guangzan** [reply]
![Video_2021-07-09_155004](https://user-images.githubusercontent.com/43458667/125043403-79202780-e0cd-11eb-9794-796b3586cceb.gif)

![image](https://user-images.githubusercontent.com/43458667/125043514-97862300-e0cd-11eb-8377-b1a888fae0b1.png)

It seems that this is the correct usage. I'll find another reason. Thanks !

**Reply by guangzan** [reply]
- vite-plugin-windicss: ^1.2.4
- vite: ^2.3.8
- windicss: ^3.1.4
- framework: vue

```ts
// windi.config.ts
import { defineConfig } from 'windicss/helpers'

export default defineConfig({
    darkMode: 'class',
})
```
`class="<md:!w-full"` doesn't work.
